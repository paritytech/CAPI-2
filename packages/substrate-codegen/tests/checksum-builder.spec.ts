import ksm from "./ksm.json"
import { expect, describe, it } from "vitest"
import { getChecksumBuilder } from "@/."

describe("getChecksumBuilder", () => {
  const builder = getChecksumBuilder(ksm as any)
  it("batched call", () => {
    const result = builder.buildCall("Utility", "batch")
    expect(result).toMatchSnapshot()
  })

  it("felloship referenda submit", () => {
    const result = builder.buildCall("FellowshipReferenda", "submit")
    expect(result).toMatchSnapshot()
  })
})
