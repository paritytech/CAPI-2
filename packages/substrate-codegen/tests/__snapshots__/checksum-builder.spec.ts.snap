// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getChecksumBuilder > batched call 1`] = `17999676605435512891n`;

exports[`getChecksumBuilder > felloship referenda submit 1`] = `6891454522755512536n`;
